sdk/python/feast/registry.py

```diff
@@ -69,54 +77,69 @@ def __init__(self, registry_path: str, repo_path: Path, cache_ttl: timedelta):
             raise Exception(
                 f"Registry path {registry_path} has unsupported scheme {uri.scheme}. Supported schemes are file and gs."
             )
-        self.cached_registry_proto_ttl = cache_ttl
+
+        if not instantiated_internally:
+            warnings.warn(
+                "The method refresh() is being deprecated since the Registry now maintains \
+                an up-to-date cache. Feast 0.12 and onwards will not support refresh().",
+                DeprecationWarning,
+            )
+
+        # Read cache from the RegistryStore.
+        self.cached_registry_proto = self._get_registry_proto()
         return
 
     def _initialize_registry(self):
-        """Explicitly forces the initialization of a registry"""
-        self._registry_store.update_registry_proto(updater=None)
+        """Explicitly initializes the registry with an empty proto."""
+        self.cached_registry_proto = RegistryProto()
+        self.cached_registry_proto.registry_schema_version = REGISTRY_SCHEMA_VERSION
 
-    def apply_entity(self, entity: Entity, project: str):
+    def apply_entity(self, entity: Entity, project: str, flush: bool = True):
         """
         Registers a single entity with Feast
 
         Args:
             entity: Entity that will be registered
             project: Feast project that this entity belongs to
+            flush: Whether the changes made should be immediately persisted
         """
         entity.is_valid()
         entity_proto = entity.to_proto()
         entity_proto.spec.project = project
+        found_entity = False
 
-        def updater(registry_proto: RegistryProto):
-            for idx, existing_entity_proto in enumerate(registry_proto.entities):
-                if (
-                    existing_entity_proto.spec.name == entity_proto.spec.name
-                    and existing_entity_proto.spec.project == project
-                ):
-                    del registry_proto.entities[idx]
-                    registry_proto.entities.append(entity_proto)
-                    return registry_proto
-            registry_proto.entities.append(entity_proto)
-            return registry_proto
+        for idx, existing_entity_proto in enumerate(
+            self.cached_registry_proto.entities
+        ):
+            if (
+                existing_entity_proto.spec.name == entity_proto.spec.name
+                and existing_entity_proto.spec.project == project
+            ):
+                found_entity = True
+                del self.cached_registry_proto.entities[idx]
+                self.cached_registry_proto.entities.append(entity_proto)
+                break
+
+        if not found_entity:
+            self.cached_registry_proto.entities.append(entity_proto)
```

Comment:
how about simplifying this to something like:

```
    for ...
        if blah:
                del self.cached_registry_proto.entities[idx]
                break


    self.cached_registry_proto.entities.append(entity_proto)
```

Reply:
Same for other things (views, tables, etc)

Reply:
Good point, I'll make this fix.